File: scratch/replace_dates.js
const fs = require('fs');

const path = "c:/Users/vidal/OneDrive/Anti-Gravity/Centinela 2.0 - copia (2)/src/pages/CompanyDashboard.jsx";
let content = fs.readFileSync(path, 'utf8');

const func_code = `
// Helper to get Argentina local date string YYYY-MM-DD
const getARDateStr = (dateInput) => {
  if (!dateInput) return '';
  try {
    const d = new Date(dateInput);
    if (isNaN(d.getTime())) return typeof dateInput === 'string' ? dateInput.split('T')[0] : '';
    // Ajustar a UTC-3. Restamos 3 horas al UTC.
    const arDate = new Date(d.getTime() - (3 * 60 * 60 * 1000));
    return arDate.toISOString().split('T')[0];
  } catch (e) {
    return String(dateInput).split('T')[0];
  }
};
`;

if (!content.includes('const getARDateStr = (dateInput) => {')) {
    content = content.replace("const CompanyDashboard = () => {", func_code + "\nconst CompanyDashboard = () => {");
}

content = content.split("new Date().toISOString().split('T')[0]").join("getARDateStr(new Date())");
content = content.split("e.fechaRegistro.split('T')[0]").join("getARDateStr(e.fechaRegistro)");
content = content.split("mediaModal.event?.fechaRegistro?.split('T')[0]").join("getARDateStr(mediaModal.event?.fechaRegistro)");

fs.writeFileSync(path, content, 'utf8');
console.log("Dates logic swapped for Argentina Time successfully.");

File: scratch/replace_dates.py
import re

path = r"c:\Users\vidal\OneDrive\Anti-Gravity\Centinela 2.0 - copia (2)\src\pages\CompanyDashboard.jsx"
with open(path, "r", encoding="utf-8") as f:
    content = f.read()

func_code = """
// Helper to get Argentina local date string YYYY-MM-DD
const getARDateStr = (dateInput) => {
  if (!dateInput) return '';
  try {
    const d = new Date(dateInput);
    if (isNaN(d.getTime())) return typeof dateInput === 'string' ? dateInput.split('T')[0] : '';
    // Ajustar a UTC-3. Restamos 3 horas al UTC.
    const arDate = new Date(d.getTime() - (3 * 60 * 60 * 1000));
    return arDate.toISOString().split('T')[0];
  } catch (e) {
    return String(dateInput).split('T')[0];
  }
};
"""

if "const getARDateStr = (dateInput) => {" not in content:
    content = content.replace("const CompanyDashboard = () => {", func_code + "\nconst CompanyDashboard = () => {")

content = content.replace("new Date().toISOString().split('T')[0]", "getARDateStr(new Date())")
content = content.replace("e.fechaRegistro.split('T')[0]", "getARDateStr(e.fechaRegistro)")
content = content.replace("mediaModal.event?.fechaRegistro?.split('T')[0]", "getARDateStr(mediaModal.event?.fechaRegistro)")

with open(path, "w", encoding="utf-8") as f:
    f.write(content)

print("Dates logic swapped for Argentina Time successfully.")
